versioned/persist/mongodb/src/main/java/org/projectnessie/versioned/persist/mongodb/MongoDatabaseAdapterFactory.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.mongodb;
+
+import org.projectnessie.versioned.persist.adapter.DatabaseAdapter;
+import org.projectnessie.versioned.persist.adapter.DatabaseAdapterFactory;
+
+public class MongoDatabaseAdapterFactory
+    implements DatabaseAdapterFactory<MongoDatabaseAdapterConfig> {
+  @Override
+  public String getName() {
+    return "MongoDB";
+  }
+
+  @Override
+  public Builder<MongoDatabaseAdapterConfig> newBuilder() {
+    return new Builder<MongoDatabaseAdapterConfig>() {
+      @Override
+      protected MongoDatabaseAdapterConfig getDefaultConfig() {
+        return ImmutableMongoDatabaseAdapterConfig.builder()
+            .connectionString("unknown")
```

Comment:
why do we need those `unknown`s?

Reply:
I did not want to make the optional in the "immutable" config since they are actually required for any connection.

Reply:
not if the connection-instance is passed ;)

Reply:
Yeah, but then we have a logical conflict between different ways of defining the connection in the same config object, don't we?

Reply:
It having these defaults would prevent checks in `MongoDatabaseClient` whether the properties are set though, right?

Reply:
I've made Mongo config properties nullable to match the pattern on other adapters: 4ba74c3889b28dcb70c6b3fb4d9a3048c92f2095